core/src/main/java/org/sonarsource/sonarlint/core/container/standalone/StandaloneGlobalContainer.java

```diff
@@ -0,0 +1,130 @@
+import org.sonarsource.sonarlint.core.container.global.GlobalTempFolderProvider;
+import org.sonarsource.sonarlint.core.container.standalone.rule.StandaloneRuleRepositoryContainer;
+import org.sonarsource.sonarlint.core.plugin.DefaultPluginJarExploder;
+import org.sonarsource.sonarlint.core.plugin.DefaultPluginRepository;
+import org.sonarsource.sonarlint.core.plugin.PluginClassloaderFactory;
+import org.sonarsource.sonarlint.core.plugin.PluginDownloader;
+import org.sonarsource.sonarlint.core.plugin.PluginLoader;
+import org.sonarsource.sonarlint.core.plugin.cache.PluginCacheProvider;
+
+public class StandaloneGlobalContainer extends GlobalContainer {
+
+  private Rules rules;
+  private ActiveRules activeRules;
+  private Context rulesDefinitions;
+
+  public static StandaloneGlobalContainer create(GlobalConfiguration globalConfig) {
+    StandaloneGlobalContainer container = new StandaloneGlobalContainer();
+    container.add(globalConfig);
+    container.add(new StandalonePluginIndexProvider(globalConfig.getPluginUrls()));
+    return container;
+  }
+
+  @Override
+  protected void doBeforeStart() {
+    add(
+      DefaultPluginRepository.class,
+      PluginDownloader.class,
+      PluginLoader.class,
+      PluginClassloaderFactory.class,
+      DefaultPluginJarExploder.class,
+      ExtensionInstaller.class,
+
+      new GlobalTempFolderProvider(),
+      UriReader.class,
+      new PluginCacheProvider(),
+      System2.INSTANCE);
+  }
+
+  @Override
+  protected void doAfterStart() {
+    installPlugins();
+
+    loadRulesAndActiveRulesFromPlugins();
+  }
+
+  private void loadRulesAndActiveRulesFromPlugins() {
+    StandaloneRuleRepositoryContainer container = new StandaloneRuleRepositoryContainer(this);
+    container.execute();
+    rules = container.getRules();
+    activeRules = container.getActiveRules();
+    rulesDefinitions = container.getRulesDefinitions();
+  }
+
+  @Override
+  public AnalysisResults analyze(AnalysisConfiguration configuration, IssueListener issueListener) {
+    AnalysisContainer analysisContainer = new AnalysisContainer(this);
+    analysisContainer.add(configuration);
+    analysisContainer.add(issueListener);
+    analysisContainer.add(rules);
+    analysisContainer.add(activeRules);
+    DefaultAnalysisResult defaultAnalysisResult = new DefaultAnalysisResult();
+    analysisContainer.add(defaultAnalysisResult);
+    analysisContainer.execute();
+    return defaultAnalysisResult;
+  }
+
+  @Override
+  public RuleDetails getRuleDetails(String ruleKeyStr) {
+    RuleKey ruleKey = RuleKey.parse(ruleKeyStr);
+    Rule rule = rules.find(ruleKey);
+    if (rule == null) {
+      throw new IllegalArgumentException("Unable to find rule with key " + ruleKey);
+    }
+    Repository repo = rulesDefinitions.repository(rule.key().repository());
+
+    return new DefaultRuleDetails(rule.name(), rule.description(), rule.severity(), repo.language(), repo.rule(rule.key().rule()).tags());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'rule' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'repo' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)